### PR TITLE
API-351: Instantly revoke access token on client delete

### PR DESCRIPTION
**Pull request by momoss**
[//]: <> (<3 Thanks for taking the time to contribute! You're awesome! <3)

[//]: <> (If you've never contributed to this repository before, please read https://github.com/akeneo/pim-community-dev/blob/master/.github/CONTRIBUTING.md)

**Description (for Contributor and Core Developer)**

[//]: <> (What does this Pull Request do? reference the related issue?)

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | -
| Added Behats                      | -
| Added integration tests           | OK
| Changelog updated                 | -
| Review and 2 GTM                  | OK
| Micro Demo to the PO (Story only) | OK
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: src/Pim/Bundle/ApiBundle/tests/integration/Client/RevokeTokenIntegration.php [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+    public function testCascadeDeleteRefreshToken()
+    {
+        $client = static::createClient();
+        list($clientId, $secret) = $this->createOAuthClient();
+
+        $client->request('POST', 'api/oauth/v1/token',
+            [
+                'username'   => static::USERNAME,
+                'password'   => static::PASSWORD,
+                'grant_type' => 'password',
```
Comment (skeleton) [inline src/Pim/Bundle/ApiBundle/tests/integration/Client/RevokeTokenIntegration.php]: should not be refresh_token ?

File: src/Pim/Bundle/ApiBundle/tests/integration/Client/RevokeTokenIntegration.php [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+        $client->request('GET', '/api/rest/v1/currencies/eur');
+        $response = $client->getResponse();
+
+        $this->assertSame(Response::HTTP_OK, $response->getStatusCode());
+
+        $this->revokeOAuthClient($clientId);
+
+        $client->request('GET', '/api/rest/v1/currencies/eur');
+        $response = $client->getResponse();
+        $responseBody = json_decode($response->getContent(), true);
```
Comment (skeleton) [inline src/Pim/Bundle/ApiBundle/tests/integration/Client/RevokeTokenIntegration.php]: do not use json_decode to test your response. test the response directly.
json_decode can impact the structure of the response (like transform an object to an array). it cannot be the case here but take the habitude please.

File: src/Pim/Bundle/ApiBundle/tests/integration/Client/RevokeTokenIntegration.php [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+                'PHP_AUTH_PW'   => $secret,
+            ]
+        );
+
+        $response = $client->getResponse();
+        $this->assertSame(Response::HTTP_OK, $response->getStatusCode());
+
+        $this->revokeOAuthClient($clientId);
+
+        $client->request('POST', '/api/oauth/v1/token',
```
Comment (skeleton) [inline src/Pim/Bundle/ApiBundle/tests/integration/Client/RevokeTokenIntegration.php]: I would add a test with an url different of the token. What's happen if user ask "/apu/rest/v1/products" route ?

File: src/Pim/Bundle/ApiBundle/tests/integration/ApiTestCase.php [hunk 4]
```diff
@@ -89,6 +89,29 @@ protected function createAuthenticatedClient(
     }
 
     /**
+     * Revoke a client using command line.
+     *
+     * @param $clientId
+     *
+     * @return string
+     */
+    protected function revokeOAuthClient($clientId)
```
Comment (ahocquard) [inline src/Pim/Bundle/ApiBundle/tests/integration/ApiTestCase.php]: move it in RevokeTokenIntegration.php as it's not used

File: src/Pim/Bundle/ApiBundle/tests/integration/Client/RevokeTokenIntegration.php (hunk 1 above)
Comment (momoss) [inline src/Pim/Bundle/ApiBundle/tests/integration/Client/RevokeTokenIntegration.php]: We should authenticate with password first time

File: src/Pim/Bundle/ApiBundle/tests/integration/Client/RevokeTokenIntegration.php (hunk 3 above)
Comment (momoss) [inline src/Pim/Bundle/ApiBundle/tests/integration/Client/RevokeTokenIntegration.php]: It's the test just before ;-)

Comment (skeleton) [inline src/Pim/Bundle/ApiBundle/tests/integration/Client/RevokeTokenIntegration.php]: sorry ^^

**Pull request merged**